ansible/test_searchengine_indices.yml

```diff
@@ -0,0 +1,19 @@
+- name: test search engine indexing, compare results come from searchengine and database server
+  hosts: "{{ idr_environment | default('idr') }}-searchengine-hosts"
+
+  tasks:
+
+   - name: Get data from postgres database and insert  them to Elasticsearch index using docker searchengine
+     become: yes
+     docker_container:
+       image: "{{ searchengine_docker_image }}"
+       name: searchengine_index_test
+       cleanup: True
+       #auto_remove: yes
+       command: "test_indexing_search_query"
```

Comment:
What's the expectation from this playbook? The execution completes but the Docker container stops immediately.

Does that mean the process completed without error or is there a way to check that the tests performed as expected?

Reply:
Thanks for pushing the image. 

```
[sbesson@test107-searchengine ~]$ sudo docker ps -a | grep index_test
e173db26ba5e   a9e7cc535ea7                                           "bash run_app.sh tes…"   4 hours ago   Exited (0) 4 hours ago                                                         searchengine_index_test
[sbesson@test107-searchengine ~]$ sudo docker logs e173db26ba5e -n 20
[2022-06-13 16:08:23,152] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,152] INFO in manage: Results for name: Imaging Method, value: flim is: OK!, 
 database servere query time= 0:00:00.045564, searchengine query time= 0:00:00.026975
[2022-06-13 16:08:23,152] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,152] INFO in manage: Results for name: name, value: idr0027-dickerson-chromatin/experimentA is: OK!, 
 database servere query time= 0:00:00.036469, searchengine query time= 0:00:00.032670
[2022-06-13 16:08:23,152] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,152] INFO in manage: Results for name: Screen Type, value: validation screen is: OK!, 
 database servere query time= 0:00:00.476731, searchengine query time= 0:00:00.099920
[2022-06-13 16:08:23,152] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,152] INFO in manage: Results for query_image_and name: query_image_and, value: [['Cell Cycle Phase', 'anaphase'], ['organism', 'homo sapiens']] is: OK!, 
 database servere query time= 0:01:25.640636, searchengine query time= 0:00:00.018535
[2022-06-13 16:08:23,152] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,153] INFO in manage: Results for query_image_or name: query_image_or, value: [['Organism Part', 'Prostate'], ['Organism Part Identifier', 'T-77100']] is: OK!, 
 database servere query time= 0:00:30.377843, searchengine query time= 0:00:00.121028
[2022-06-13 16:08:23,153] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,153] INFO in manage: Results for query_image_and_or name: query_image_and_or, value: {'query_image_or': [['Organism Part', 'Prostate'], ['Organism Part Identifier', 'T-77100']], 'query_image_and': [['Organism', 'Homo sapiens'], ['Antibody Identifier', 'CAB034889']]} is: OK!, 
 database servere query time= 0:02:00.537367, searchengine query time= 0:00:00.015820
[2022-06-13 16:08:23,153] INFO in manage: -----------------------------------------------------------------------------
[2022-06-13 16:08:23,153] INFO in manage: ###########################################################################################################
```

A few questions/comments:

- the output generally looks good. Would the return code be non-zero if there is any mismatch? Is the output saved anywhere else than Docker logs
- I am concerned about the addition of playbooks especially as they might not get executed in a standard provisioning workflow. Since this is effectively validating the index generation , could this command be calledas the last step of  `get_index_data_from_database`?
- what is the roadmap/timeline towards tagging the code base and the Docker image ?